sdk/storage/storage-file-share/swagger/README.md

```diff
@@ -246,4 +248,28 @@ directive:
       $.Expiry["x-ms-client-name"] = "expiresOn";
 ```
 
+### Update date type for "fileCreatedOn" and "fileLastWriteOn" from `DateTimeRfc1123` -> `string`
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $.parameters.FileCreationTime
+    transform: >
+      delete $.format;
+  - from: swagger-document
+    where: $.parameters.FileLastWriteTime
+    transform: >
+      delete $.format;
+```
+
+### Retain XStore swagger behavior - Revert file permissions content-type to "application/xml" from "application/json"(unified swagger)
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["x-ms-paths"]["/{shareName}?restype=share&comp=filepermission"]
+    transform: >
+      $.put.consumes = ["application/xml"];
```

Comment:
This contentType attribute `contentType: "application/xml; charset=utf-8",`  of `createPermissionOperationSpec` from the XStore swagger is deleted in the new codegen from unified swagger. 
   
https://github.com/Azure/azure-sdk-for-js/pull/5980/commits/d03b325935246e16eebf75338d7da26013485463#diff-ea248212a35437cb5ab4aee596288febR439

API is broken as a consequence.  

----------------------------------------------
**More info on swaggers**
In the global scope of the swagger, `consumes` is set to "application/xml" in both the xstore swagger as well as the unified swagger.
```yaml
    "consumes": [
        "application/xml"
    ]
```
The only difference in the unified swagger is that the `Share_CreatePermission` operation additionally overrides it to `"application/json"`.

**Share_CreatePermission - XStore swagger vs unified swagger**
![image](https://user-images.githubusercontent.com/10452642/68888756-6410ac80-06d0-11ea-86ca-bc73be41d053.png)

All I did was reverting this with a transform to retain the XStore swagger's behaviour.

---------------------------------------------

I think .NET SDK tries to do a similar thing of defaulting it to "application/xml" with `delete $.put.consumes;` here - [Link to the swagger transform](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage/Azure.Storage.Files.Shares/swagger#temporarily-work-around-proper-json-support-for-file-permissions)

Tried to do the same `delete $.put.consumes;` just like .NET but it didn't seem to work.

Instead, adding this transform - `$.put.consumes = ["application/xml"];` retained the `contentType: "application/xml; charset=utf-8",`   that used to be present in the codegen of XStore swagger which seems to fix the API and the tests succeeded.

Reply:
Thanks for the asking, @tg-msft.  
This contentType attribute `contentType: "application/xml; charset=utf-8",`  of `createPermissionOperationSpec` from the XStore swagger is deleted in the new codegen from unified swagger. 
   
https://github.com/Azure/azure-sdk-for-js/pull/5980/commits/d03b325935246e16eebf75338d7da26013485463#diff-ea248212a35437cb5ab4aee596288febR439

API is broken as a consequence.  

----------------------------------------------
**More info on swaggers**
In the global scope of the swagger, `consumes` is set to "application/xml" in both the xstore swagger as well as the unified swagger.
```yaml
    "consumes": [
        "application/xml"
    ]
```
The only difference in the unified swagger is that the `Share_CreatePermission` operation additionally overrides it to `"application/json"`.

**Share_CreatePermission - XStore swagger vs unified swagger**
![image](https://user-images.githubusercontent.com/10452642/68888756-6410ac80-06d0-11ea-86ca-bc73be41d053.png)

All I did was reverting this with a transform to retain the XStore swagger's behaviour.

---------------------------------------------

I think .NET SDK tries to do a similar thing of defaulting it to "application/xml" with `delete $.put.consumes;` here - [Link to the swagger transform](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage/Azure.Storage.Files.Shares/swagger#temporarily-work-around-proper-json-support-for-file-permissions)

Tried to do the same `delete $.put.consumes;` just like .NET but it didn't seem to work.

Instead, adding this transform - `$.put.consumes = ["application/xml"];` retained the `contentType: "application/xml; charset=utf-8",`   that used to be present in the codegen of XStore swagger which seems to fix the API and the tests succeeded.
Let me know if I'm missing something or if there is a better workaround.

Reply:
.NET was removing it because the Get/CreatePermission APIs [expect JSON.](https://docs.microsoft.com/en-us/rest/api/storageservices/get-permission#response-body)  By removing it, our generator switched to using strings which we could [build and parse from JSON ourselves.](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs#L1402)  I just wonder how this works in JS if your generated code is trying to parse XML instead?

Reply:
@jeremymeng

Reply:
@HarshaNalluru What happens to JS sdk if keeping the "overrides it to "application/json"."?

Reply:
Is this related to the `isXml: true` bug in autorest typescript plugin?  We had to patched it to `isXml: false` after running the codegen.

Reply:
@jeremyment Manually path `isXml: false` after every codegen? The bug should be fixed in autorest typescript plugin.

Reply:
Agree that we should fix the autorest typescript plugin. I found the issue: https://github.com/Azure/autorest.typescript/issues/465. The patch is a manual workaround before we have a fix.

Reply:
> Is this related to the isXml: true bug in autorest typescript plugin?

@jeremymeng Manual update to `isXML: false` seems to be unrelated to this "application/json" vs "application/xml" issue that I mentioned here - https://github.com/Azure/azure-sdk-for-js/pull/5980#discussion_r346508077.

> @HarshaNalluru What happens to JS sdk if keeping the "overrides it to "application/json"."?

@XiaoningLiu, createPermission throws an error as follows.

```bash
  1) ShareClient
       create and get permission:
     Error: ﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:9c453f06-001a-005f-49f0-9e9402000000
Time:2019-11-19T15:47:15.2089521Z</Message><AuthenticationErrorDetail>The MAC signature found in the HTTP request 'K6g1aun8EeenNtGArwcTkd15dnm5FYS0rq5Qijuiaxw=' is not the same as any computed signature. Server used following string to sign: 'PUT
247
text/plain;charset=UTF-8
x-ms-client-request-id:ebe2e059-5ee5-4d5b-9576-0745ebba57c0
x-ms-date:Tue, 19 Nov 2019 23:47:14 GMT
x-ms-version:2019-02-02
/harshanstoragetest/share157420723124200350
comp:filepermission
restype:share'.</AuthenticationErrorDetail></Error>
      at new RestError (<path>\azure-sdk-for-js\sdk\core\core-http\lib\restError.ts:23:5)
      at <path>\sdk\core\core-http\lib\policies\deserializationPolicy.ts:168:27
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

Reply:
Something doesn't sound right here that we need both `contentType` of `application/xml` and `isXml: false` at the same time. But I think we should use @HarshaNalluru's workaround now to merge this PR and investigate later.

Reply:
I agree.
contentType of `application/xml` and `isXml: false` are present from the beginning (XStore swagger). This PR adds a transform for the unified swagger just to retain the old behaviour.
We can further investigate the underlying issue.

Reply:
I tried a bit more. Here's what I've found so far

1. when `"consumes"` is set to `[ "application/json" ]`, the operation spec will not have the `contentType` field. I guess that json is the default so it is omitted.

2. however, without the contentType field, core-http seems to use content type of `text/plain;charset=UTF-8` for the request, which leads to HMAC signature verification failure.

3. Changing the `contentType` field in `createPermissionOperationSpec` to `application/json; charset=utf-8` also make the particular test to pass.

It is still a mystery that using `applicaton/xml` also worked.  I will log a new issue to investigate further.  I think the workaround in this PR is fine.

Reply:
Logged https://github.com/Azure/autorest.typescript/issues/500